parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -61,10 +61,10 @@ public LegionBoardParser(SubstitutionScheduleData scheduleData, CookieProvider c
 		} catch (JSONException e) {
 			e.printStackTrace();
 		}
-		authKey = getAuthenticationKey(getCredential());
 	}
 
 	public SubstitutionSchedule getSubstitutionSchedule() throws IOException, JSONException, CredentialInvalidException {
+		authKey = getAuthenticationKey(getCredential());
```

Comment:
Ah, that's true, it doesn't work in the constructor because the credential is set using `setCredential(Credential credential)` afterwards. To still make it work when `getAllClasses` (or Teachers) is called independently, you could also just drop the `authKey` field and just use `getAuthenticationKey` everywhere. Or override the `setCredential(Credential credential)` method to update the `authKey` field there.